server/modules/search/elasticsearch/engine.js

```diff
@@ -87,63 +84,156 @@ module.exports = {
    * @param {Object} opts Additional options
    */
   async query(q, opts) {
-    try {
-      const results = await this.client.search({
-        index: this.config.indexName,
-        body: {
-          query: {
-            bool: {
-              filter: [
-                {
-                  bool: {
-                    should: [
-                      {
-                        simple_query_string: {
-                          query: q
-                        }
-                      },
-                      {
-                        query_string: {
-                          query: "*" + q + "*"
-                        }
-                      }
-                    ],
-                    minimum_should_match: 1
+    if(q.search("date:")==0){
+      // get the date range
+      if (q.length == 26){
+        var start_date = q.substr(5,10);
+        var stop_date = q.substr(16,10);
```

Comment:
Avoid using exact character indexes. Check instead of the presence of a comma and split the string if present. This allows for greater flexibility in the format.

Also, the ISO 8601 standard for date range uses a `/` as a separator.

```suggestion
      if (q.indexOf(',') > 0) {
        const dateRange = q.substring(5).split('/')
```

Finally, never use `var`, always `let` or `const`. `var` is globally scoped and unsafe.